### PR TITLE
Create modify-compressed-data.md

### PR DESCRIPTION
**Pull request by antekresic**
# Description

[Short summary of why you created this PR]

# Links

Fixes #[insert issue link, if any]

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/about/latest/contribute-to-docs/)

# Review checklists

Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

*   [ ] Is the content technically accurate?
*   [ ] Is the content complete?
*   [ ] Is the content presented in a logical order?
*   [ ] Does the content use appropriate names for features and products?
*   [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

*   [ ] Is the content free from typos?
*   [ ] Does the content use plain English?
*   [ ] Does the content contain clear sections for concepts, tasks, and references?
*   [ ] Have any images been uploaded to the correct location, and are resolvable?
*   [ ] If the page index was updated, are redirects required
      and have they been implemented?
*   [ ] Have you checked the built version of this content?

### CODE REVIEW

File: api/compression.md [hunk 1]
```diff
@@ -21,13 +21,8 @@ manually][compress_chunk], instead of automatically as they age.
...
 *   `ROW LEVEL SECURITY` is not supported.
-*   Constraints are not fully supported:
-    *   Unique constraints (`UNIQUE`) are restricted to compression `SEGMENTBY`
-      and `ORDER BY` columns.
-    *   Primary Keys (`PRIMARY KEY`) are restricted to compression `SEGMENTBY`
-      and `ORDER BY` columns.
-    *   Constraints are restricted to compression `SEGMENTBY` columns when
-    applicable. Otherwise they are unsupported.
+*   Creation of unique constraints on compressed chunks is not supported
+   *   Its possible to add them by disabling compression and re-enabling after constraint creation.
```
Comment (Loquacity) [inline api/compression.md]:
```suggestion
   *   You can add them by disabling compression and re-enabling after constraint creation.
```

File: api/compression.md [hunk 2]
```diff
@@ -41,6 +36,7 @@ based on the version of TimescaleDB you are currently running.
 | 1.5 - 2.0 | Data and schema modifications are not supported. |
 | 2.1 - 2.2 | Schema may be modified on compressed hypertables. Data modification not supported. |
 | 2.3 | Schema modifications and basic insert of new data are allowed. Deleting, updating and some advanced insert statements are not supported. |
+| 2.11 | Deleting, updating and advanced insert statements are supported. |
```
Comment (Loquacity) [inline api/compression.md:39]: All tables should be unspaced, but it's not a deal breaker.

**Review by Loquacity — CHANGES_REQUESTED** [review]

File: use-timescale/compression/modify-compressed-data.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+---
+title: Inserting or modifying data in compressed chunks
+excerpt: What happens when you try to modify data in a compressed hypertable
+products: [cloud, mst, self_hosted]
+keywords: [compression, backfilling, hypertables]
+---
+
+# Inserting into compressed chunks
+
+Prior to version 2.10, inserting into compressed chunks was blocked due to certain
```
[the remark this answers is not in the document]
Comment (antekresic) [reply use-timescale/compression/modify-compressed-data.md:10]: Please make it more readable/understandable, writing is not my best quality :sweat_smile:

**Review by iroussos — CHANGES_REQUESTED** [review]
Thank you @antekresic for creating this initial update to the compression docs!

@Loquacity I am a little bit torn here, so I would love your opinion and feedback on how to iterate on this:

On one side, I think that we should get out at least something as 2.11 is already out in the wild and our current docs are no longer valid.

But I think that we should also dive more into the details and gotchas of compression, similar to what we have done on our [enablement document](https://docs.google.com/document/d/1vwz4UmjwUFlKk3ORDCeyX2vczrIGOWREmNu0y1oM-Ec/edit) and presentation ([slides](https://docs.google.com/presentation/d/1ykMlhV355bJIoU1YunnImTiBafIfLvJ_cUmBbXnDtIQ/edit?usp=sharing), [recording](https://timescale.zoom.us/rec/share/WZwOeZ1ng3D5OeLI9FVVMwPvAqMb4JAKwYRPuzU-Vss6JF8Ytb8_PxvoxhLKZnUf.nQzWAvyivCY3K0X8?pwd=uPTvxEjlJmw_hShNHKn6n2woURE7CnXe)).

I would propose that we should at least bring our documentation on par with the [Best practices & Gotchas section of the enablement document](https://docs.google.com/document/d/1vwz4UmjwUFlKk3ORDCeyX2vczrIGOWREmNu0y1oM-Ec/edit#heading=h.4fyanohzulxw).

@Loquacity should we wrap this PR, fix the few issues with the new md and get it out and then work together on an enhanced version? You can check the enablement doc and my presentation and we can then think about how to approach it together.

**Comment by Loquacity** [comment]
> Thank you @antekresic for creating this initial update to the compression docs!
> 
> @Loquacity I am a little bit torn here, so I would love your opinion and feedback on how to iterate on this:
> 
> On one side, I think that we should get out at least something as 2.11 is already out in the wild and our current docs are no longer valid.
> 
> But I think that we should also dive more into the details and gotchas of compression, similar to what we have done on our [enablement document](https://docs.google.com/document/d/1vwz4UmjwUFlKk3ORDCeyX2vczrIGOWREmNu0y1oM-Ec/edit) and presentation ([slides](https://docs.google.com/presentation/d/1ykMlhV355bJIoU1YunnImTiBafIfLvJ_cUmBbXnDtIQ/edit?usp=sharing), [recording](https://timescale.zoom.us/rec/share/WZwOeZ1ng3D5OeLI9FVVMwPvAqMb4JAKwYRPuzU-Vss6JF8Ytb8_PxvoxhLKZnUf.nQzWAvyivCY3K0X8?pwd=uPTvxEjlJmw_hShNHKn6n2woURE7CnXe)).
> 
> I would propose that we should at least bring our documentation on par with the [Best practices & Gotchas section of the enablement document](https://docs.google.com/document/d/1vwz4UmjwUFlKk3ORDCeyX2vczrIGOWREmNu0y1oM-Ec/edit#heading=h.4fyanohzulxw).
> 
> @Loquacity should we wrap this PR, fix the few issues with the new md and get it out and then work together on an enhanced version? You can check the enablement doc and my presentation and we can then think about how to approach it together.

Let me check out this branch and give it an edit. I'll read over the enablement docs at the same time, and see what else we need to add, then I'll get back to you. By end of week, promise!

File: use-timescale/compression/modify-compressed-data.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+products: [cloud, mst, self_hosted]
+keywords: [compression, backfilling, hypertables]
+---
+
+# Inserting into compressed chunks
+
+In TimescaleDB&nbsp;2.11 and later, you can insert data into compressed chunks.
+This works even if the data you are inserting has unique constraints, and those
+constraints are preserved during the insert operation. This is done by using a
+PostgreSQL function that decompresses relevant data on the fly to check if the
```
Comment (Rajakavitha1) [inline use-timescale/compression/modify-compressed-data.md:13]:
```suggestion
PostgreSQL function that decompresses relevant data during the insert to check if the
```

File: use-timescale/compression/modify-compressed-data.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+compressed chunk, a small amount of data is decompressed to allow a speculative
+insertion, and block any inserts which could violate constraints.
+
+In TimescaleDB&nbsp;2.10, you can insert data into compressed chunks with some
+limitations. The primary limitation is that you can't insert data with unique
+constraints. Additionally, newly inserted data needs to be compressed at the
+same time as the data in the chunk, either by a running recompression policy, or
+by using `recompress_chunk` manually on the chunk.
+
+In TimescaleDB&nbsp;2.9 and earlier, you cannot insert data into compressed chunks.
```
Comment (Rajakavitha1) [inline use-timescale/compression/modify-compressed-data.md:24]: Just wondering if the different version information can be best presented in the Tabs <Version 2.11 and later> ,<Version 2.10>, <Version 2.9 and earlier>

**Review by Rajakavitha1 — APPROVED** [review]
LGTM!!!

File: use-timescale/compression/modify-compressed-data.md (hunk 5 above)
Comment (Loquacity) [reply use-timescale/compression/modify-compressed-data.md:24]: hrm, that's a good idea. Let me try it and we can see how it looks.

**Comment by Loquacity** [comment]
image

I quite like this! @Rajakavitha1 and @antekresic what do you think?

**Comment by antekresic** [comment]
I like it, it focuses on the latest version which is good IMO.

**Comment by Loquacity** [comment]
@antekresic I'm happy to merge this, if you are ... ?

**Review by sb230132 — APPROVED** [review]

**Review by Loquacity — APPROVED** [review]

**Pull request merged**